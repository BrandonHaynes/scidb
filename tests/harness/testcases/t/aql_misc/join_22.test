# ticket no. 1041

--setup
--start-query-logging
create array A <a:string> [x=-2:3,2,1,y=0:2,3,0]
create array B <a:int32> [x=-2:3,2,1,y=0:2,3,0]

--test
--aql select * into C from build(A,x),build(B,x)
--aql select * into D from C,C
--aql select * from show(D)
--aql select * from D where a<>a_3
--aql select * from D where a=a_3

--cleanup
remove(A)
remove(B)
remove(C)
remove(D)
--stop-query-logging
