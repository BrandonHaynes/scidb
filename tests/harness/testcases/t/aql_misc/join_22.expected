SCIDB QUERY : <create array A <a:string> [x=-2:3,2,1,y=0:2,3,0]>
Query was executed successfully

SCIDB QUERY : <create array B <a:int32> [x=-2:3,2,1,y=0:2,3,0]>
Query was executed successfully

SCIDB QUERY : <select * into C from build(A,x),build(B,x)>
{x,y} a,a_2
{-2,0} '-2',-2
{-2,1} '-2',-2
{-2,2} '-2',-2
{-1,0} '-1',-1
{-1,1} '-1',-1
{-1,2} '-1',-1
{0,0} '0',0
{0,1} '0',0
{0,2} '0',0
{1,0} '1',1
{1,1} '1',1
{1,2} '1',1
{2,0} '2',2
{2,1} '2',2
{2,2} '2',2
{3,0} '3',3
{3,1} '3',3
{3,2} '3',3

SCIDB QUERY : <select * into D from C,C>
{x,y} a,a_2,a_3,a_2_2
{-2,0} '-2',-2,'-2',-2
{-2,1} '-2',-2,'-2',-2
{-2,2} '-2',-2,'-2',-2
{-1,0} '-1',-1,'-1',-1
{-1,1} '-1',-1,'-1',-1
{-1,2} '-1',-1,'-1',-1
{0,0} '0',0,'0',0
{0,1} '0',0,'0',0
{0,2} '0',0,'0',0
{1,0} '1',1,'1',1
{1,1} '1',1,'1',1
{1,2} '1',1,'1',1
{2,0} '2',2,'2',2
{2,1} '2',2,'2',2
{2,2} '2',2,'2',2
{3,0} '3',3,'3',3
{3,1} '3',3,'3',3
{3,2} '3',3,'3',3

SCIDB QUERY : <select * from show(D)>
{i} schema
{0} 'D<a:string,a_2:int32,a_3:string,a_2_2:int32> [x=-2:3,2,1,y=0:2,3,0]'

SCIDB QUERY : <select * from D where a<>a_3>
{x,y} a,a_2,a_3,a_2_2

SCIDB QUERY : <select * from D where a=a_3>
{x,y} a,a_2,a_3,a_2_2
{-2,0} '-2',-2,'-2',-2
{-2,1} '-2',-2,'-2',-2
{-2,2} '-2',-2,'-2',-2
{-1,0} '-1',-1,'-1',-1
{-1,1} '-1',-1,'-1',-1
{-1,2} '-1',-1,'-1',-1
{0,0} '0',0,'0',0
{0,1} '0',0,'0',0
{0,2} '0',0,'0',0
{1,0} '1',1,'1',1
{1,1} '1',1,'1',1
{1,2} '1',1,'1',1
{2,0} '2',2,'2',2
{2,1} '2',2,'2',2
{2,2} '2',2,'2',2
{3,0} '3',3,'3',3
{3,1} '3',3,'3',3
{3,2} '3',3,'3',3

SCIDB QUERY : <remove(A)>
Query was executed successfully

SCIDB QUERY : <remove(B)>
Query was executed successfully

SCIDB QUERY : <remove(C)>
Query was executed successfully

SCIDB QUERY : <remove(D)>
Query was executed successfully

